lib/vagrant-libvirt/config.rb

```diff
@@ -538,6 +538,15 @@ def _generate_uri
       end
 
       def finalize!
+        @uri = ENV['VAGRANT_LIBVIRT_URI'] if (@uri == UNSET_VALUE && ENV['VAGRANT_LIBVIRT_URI'])
+        if (@uri && @uri != UNSET_VALUE)
+          uri = URI(@uri)
+          @driver = u.scheme.sub('+ssh','') if @driver == UNSET_VALUE
```

Comment:
Ha, thanks for spotting the typo!

So would you suggest not to override the driver from the URI at all?